docs/topics/running/environment.md

```diff
@@ -62,32 +62,71 @@ verbose, valid log levels are `error`, `warn`/`warning`, `info`,
-Stack
-container](../../using/filters/#installing-self-signed-certificates).
-
-Note that when using a port name instead of a port number in a Go
-network address (as as in `REDIS_URL` or `REDIS_PERSECOND_URL`), the
-name refers a well-known port name in the container's `/etc/services`,
-and **not** to a Kubernetes port name.
+#### Redis layer 4 connectivity
+
+- `SOCKET_TYPE` and `URL` are the Go network name and Go network
+  address to dial to talk to Redis; see [Go `net.Dial`][].  Note that
+  when using a port name instead of a port number, the name refers a
+  well-known port name in the container's `/etc/services`, and **not**
+  to a Kubernetes port name.
+- `TLS_ENABLED` (new in 1.5.0) specifies whether to use TLS when
+  talking to Redis.
+- `TLS_INSECURE` (new in 1.5.0) specifies whether to skip certificate
+  verification when using TLS to talk to Redis.  Alternatively,
+  consider [installing Redis' self-signed certificate in to the
+  Ambassador Edge Stack
+  container](../../using/filters/#installing-self-signed-certificates)
+  in order to leave certificate verification on.
+
+#### Redis authentication
+
+- If `PASSWORD` (new in 1.5.0) is non-empty, then it is used to `AUTH`
+  to Redis immediately after the connection is established.
+- If `USERNAME` (new in 1.5.0) is set, then that username is used with
+  the password to log in as that user in the [Redis 6 ACL].  It is
+  invalid to set a username without setting a password.  It is invalid
+  to set a username with Redis 5 or lower.
+
+#### Redis performance tuning
+
+- `POOL_SIZE` is the number of connections to keep around when idle.
+  The total number of connections may go lower than this if there are
+  errors.  The total number of connections may go higher than this
+  during a load surge.
+- `PING_INTERVAL` (new in 1.6.0) Of the idle connections in the normal
+  pool (not extra connections created for a load surge), Ambassador
+  will `PING` one of them every `PING_INTERVAL` seconds; each
+  connection will on average be `PING`ed every
+  `PING_INTERVAL×POOL_SIZE` seconds; increasing `POOL_SIZE` without
+  reducing `PING_INTERVAL` will increase the amount of time between
+  `PING`s on a given connection.  (Backward incompatibility: prior to
+  the introduction of this setting in 1.6.0, it behaved as if
+  `PING_INTERVAL=10s÷POOL_SIZE`.)
+- `IO_TIMEOUT` sets 3 different timeouts:
+   1. `(*net.Dialer).Timeout` for establishing connections
+   2. `(*redis.Client).ReadTimeout` for reading a single complete response
+   3. `(*redis.Client).WriteTimeout` for writing a single complete request
+
+During a load surge, if the pool is depleted, Ambassador allows new
+connections to be created as fast as necessary for the first
+`POOL_SIZE` connections; once the number of connections reaches
+`2×POOL_SIZE` it is limited to creating only 1 new connection per
+second.  (Backward incompatibility: prior to 1.6.0 it never limited
```

Comment:
To avoid a hanging 'it', I would rewrite it as "prior to 1.6.0, new connection creation is not limited during a surge"